### PR TITLE
Avg& MaxPooling with SMPC

### PR DESCRIPTION
**Pull request by abogaziah**
trying to run models with maxpooling like resnet using pysyft SMPC

### CODE REVIEW

**Comment by abogaziah** [comment]
memory error: Process finished with exit code -1073741571 (0xC00000FD)
I don't think it's possible to run it currently, at least on my machine

**Comment by LaRiffle** [comment]
Can you provide the code you're trying to run ? :)

**Comment by abogaziah** [comment]
```
bob = sy.VirtualWorker(hook, id="bob")
alice = sy.VirtualWorker(hook, id="alice")
crypto_provider = sy.VirtualWorker(hook, id="james")

resnet = models.resnet18()
resnet.fix_prec().share(bob, alice, crypto_provider=crypto_provider)
data = torch.ones(4, 3, 32, 32).fix_prec().share(bob, alice, crypto_provider=crypto_provider)

print(resnet(data))

```

File: syft/frameworks/torch/tensors/interpreters/additive_shared.py [hunk 1]
```diff
@@ -819,11 +819,18 @@ def split(tensor_shares, *args, **kwargs):
...
-                def relu(tensor_shares):
+                def relu(tensor_shares, inplace=False):
                     return tensor_shares.relu()
 
                 module.relu = relu
 
+                @overloaded.function
+                def maxpool2d(tensor_shares, *args, **kwargs):
+                    a_sh = AdditiveSharingTensor(shares=tensor_shares)
+                    return maxpool2d(a_sh)
```
Comment (LaRiffle) [inline syft/frameworks/torch/tensors/interpreters/additive_shared.py:830]:
```suggestion
                def maxpool2d(tensor, *args, **kwargs):
                    return maxpool2d(tensor)
```

**Comment by abogaziah** [comment]
Solved #3369 in the last commits

**Comment by gmuraru** [comment]
The test is failing with:
```
 def float_prec(self):
>       if isinstance(self.child, PointerTensor):
E       AttributeError: 'Parameter' object has no attribute 'child'
```
It is possible because the plan is using a new type -- Parameter and that type do not have a child.

**Pull request closed without merging**